pkg/stargate/deployments.go

```diff
@@ -332,28 +338,62 @@ func computeHeapSize(template *api.StargateTemplate) resource.Quantity {
 	return resource.MustParse("256Mi")
 }
 
+// This config map will always be created by the k8ssandra controller.
+// It will augment the user provided config map with encryption settings if enabled.
 func computeVolumes(template *api.StargateTemplate) []corev1.Volume {
-	if template.CassandraConfigMapRef != nil {
-		return []corev1.Volume{{
-			Name: "cassandra-config",
-			VolumeSource: corev1.VolumeSource{
-				ConfigMap: &corev1.ConfigMapVolumeSource{
-					LocalObjectReference: *template.CassandraConfigMapRef,
+	volumes := []corev1.Volume{}
+	volumes = append(volumes, corev1.Volume{
+		Name: "cassandra-config",
+		VolumeSource: corev1.VolumeSource{
+			ConfigMap: &corev1.ConfigMapVolumeSource{
+				LocalObjectReference: corev1.LocalObjectReference{
+					Name: CassandraConfigMap,
 				},
 			},
-		}}
+		},
+	})
+
+	return volumes
+}
+
+func computeEncryptionVolumes(template *api.StargateTemplate, spec api.StargateSpec) []corev1.Volume {
+	volumes := []corev1.Volume{}
+
+	if spec.ServerEncryptionStores != nil {
+		serverEncryptionVolumes := cassandra.EncryptionVolumes("server", *spec.ServerEncryptionStores)
+		for _, volume := range serverEncryptionVolumes {
+			volumes = append(volumes, volume)
+		}
 	}
-	return nil
+	if spec.ClientEncryptionStores != nil {
+		clientEncryptionVolumes := cassandra.EncryptionVolumes("client", *spec.ClientEncryptionStores)
+		for _, volume := range clientEncryptionVolumes {
+			volumes = append(volumes, volume)
+		}
+	}
+
+	return volumes
 }
 
-func computeVolumeMounts(template *api.StargateTemplate) []corev1.VolumeMount {
-	if template.CassandraConfigMapRef != nil {
-		return []corev1.VolumeMount{{
-			Name:      "cassandra-config",
-			MountPath: cassandraConfigDir,
-		}}
+func computeVolumeMounts(template *api.StargateTemplate, encryptionVolumes []corev1.Volume) []corev1.VolumeMount {
+	mounts := []corev1.VolumeMount{}
+	mounts = append(mounts, corev1.VolumeMount{
+		Name:      "cassandra-config",
+		MountPath: cassandraConfigDir,
+	})
+	mounts = append(mounts, encryptionStoreMounts(encryptionVolumes)...)
+	return mounts
+}
+
+func encryptionStoreMounts(encryptionVolumes []corev1.Volume) []corev1.VolumeMount {
+	mounts := []corev1.VolumeMount{}
+	for _, volume := range encryptionVolumes {
+		mounts = append(mounts, corev1.VolumeMount{
+			Name:      volume.Name,
+			MountPath: cassandra.StoreMountFullPath(strings.Split(volume.Name, "-")[0], strings.Split(volume.Name, "-")[1]),
```

Comment:
Splitting strings sounds a bit fragile, I'd rather compute the mount path from the original `EncryptionStores` , just as you did for volumes in `computeEncryptionVolumes` above.

Reply:
Right, I'm reworking that part to remove string manipulations.